Add edit trip modal

rachelekm opened the pull request:
## Overview

Adds the trip editing modal for users to update:
1. active destination
2. the active mode used for routing an calculations, set by toggles for traffic conditions, is user has vehicle or is using commuter rail.

Currently "Edit your trips" does not open anything until #739 (any addition/removing of destinations expected to be handled in wizard).

### Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [x] Run `./scripts/format` to lint, format, and fix the application source code.
- [x] CI passes after rebase

### Demo


https://github.com/user-attachments/assets/0ac92e8b-9f00-4ae5-a37f-652dc5386a45



## Testing Instructions

 * `scripts/server`
 * Login to django admin and confirm your user has multiple destinations
 * Login to app and confirm correct active destination (first in destinations list, if none selected) in redux state
 * Click on the edit trips button in the user profile subheader
 * Confirm edit trips modal opens
 * Confirm user profile state in redux is selected by default (eg. the active destination is expanded option in accordion, the correct traffic condition is toggled ("Peak" by default), the correct commute options are selected ("car" if user.hasVehicle and commuter checkbox selected if "user.useCommuterRail).
 * Select a variety of options and confirm user profile state updates correctly when closing modal
 * Confirm the `networks.activeMode` updates after fulfilled user profile with the correct option key (now taking into account "peak" and "off-peak")
 * Confirm ranking list re-generates
 * In neighborhood details confirm commute range and "Your Trips" maps updated to match user-selected options (e.g. car instead of transit)

Resolves #728

rachelekm committed:
Add Edit trips modal

rachelekm committed:
Update logic to set activeMode state

rachelekm committed:
Update CHANGELOG

aaronxsu commented on src/frontend/src/pages/Discover/ModalsWrapper.tsx:
```diff
@@ -1,6 +1,8 @@
 import EditTripsModal from "src/components/EditTripsModal";
+import NeighborhoodDetail from "../NeighborhoodDetail";
+
+export const modalHeadingClassName = "text-lg font-bold text-gray-900";
```
Wondering why this is defined here rather than where it's used. Lemme know if there is any concern I am missing here.

rachelekm replied:
I was sharing this with other modals but ultimately it's now only between 2 components and really not significant enough to warrant being pulled out. I took this out in [856b547](https://github.com/azavea/echo-locator/pull/744/commits/856b54751d57f9d22770e688a9af8590dfced79f)

aaronxsu commented on src/frontend/src/components/EditTripsModal.tsx:
```diff
@@ -1,32 +1,198 @@
+    const profile = useAppSelector(selectUserProfile);
+    const trafficConditions = useAppSelector(selectTrafficConditions);
+    const [profileBuffer, setProfileBuffer] =
+        useState<UserProfileSliceState | null>(null);
```
For the buffer implementations in the recent PRs, I wonder, why not initializing the `profileBuffer` by `profile` directly. Did you run into performance issues or other state management issues?

rachelekm replied:
Good call out, I was just moving fast and copying from the wizard. I updated this in [856b547](https://github.com/azavea/echo-locator/pull/744/commits/856b54751d57f9d22770e688a9af8590dfced79f) for both edit modals so its directly initialized and I didn't see any issues

aaronxsu commented on src/frontend/src/components/EditTripsModal.tsx:
```diff
@@ -1,32 +1,198 @@
+                        <h2 className={modalHeadingClassName}>
+                            {t("editTripsModal.travelModeHeading")}
+                        </h2>
+                        {/* Toggles in sync with logic in StepTravelMode
```
Is this a good thing or something we should change?

rachelekm replied:
I took out this comment since its confusing, just wanted a way to reference the clarifying comment in StepTravelMode

aaronxsu commented on src/frontend/src/components/EditTripsModal.tsx:
```diff
@@ -1,32 +1,198 @@
+                        <ModalCloseButton onPress={() => onOpenChange(false)} />
+                    </div>
+                    <div className="flex flex-col gap-4">
+                        <Accordion
```
I'd personally make these destination options as radio buttons that look like the vertical button group in the design, which would likely be closer to what it achieves semantically. You might have considered it too. But that would mean a new component to be created. Given the remaining time and budget, using an accordion works too.

rachelekm replied:
Yeah good to document this decision here, because I did think about that but given our budget/an already-styled accordion this felt like the best way forward. It'd be great to not only use that new radio group component here but also for the commute group component when a user's setup for car directions or a neighborhood is too far. I'll open up a tech debt issue for us and hopefully we can sneak it in at the end

aaronxsu reviewed (approved):
Looking great and working as advertised. I left some questions in the comments. I also found two small styling issues- you might have caught them as well, but just surfacing them here in case we forget

1. If the purpose is friends and family, in the mobile view, it looks like the following

<img width="380" height="143" alt="Image" src="https://github.com/user-attachments/assets/a5a472f4-cd5b-485d-8bc5-29fd559cfaab" />

2. In the desktop view, when the address/label is long, it looks like the following

<img width="366" height="138" alt="Image" src="https://github.com/user-attachments/assets/ee936dd3-413b-46cd-aeb3-e2d1f20d5463" />

rachelekm committed:
Clean up edit modals logic

rachelekm committed:
Clean up mobile edit header buttons

rachelekm commented:
Thanks, @aaronxsu! I implemented your suggestions (all except for the accordion -> radio group). The edit button styling was partially addressed in a previous PR (https://github.com/azavea/echo-locator/pull/742/commits/8f0be27f75d4fd443656b7424195a470d35c5a28) but didn't fix the purpose overflow in mobile view. That's fixed in  [b70316f](https://github.com/azavea/echo-locator/pull/744/commits/b70316f6e0b5cafdc54b772b7571edd6726ad2d2).
<img width="295" height="279" alt="Screenshot 2025-10-22 at 10 47 35 AM" src="https://github.com/user-attachments/assets/75c836e4-90bf-49c8-b756-b334bff603a6" />

<img width="378" height="175" alt="Screenshot 2025-10-22 at 10 55 56 AM" src="https://github.com/user-attachments/assets/fc950679-4d68-492f-b52a-4048be3659a8" />

I also rebased on develop and will go ahead and merge on successful CI.

aaronxsu commented on src/frontend/src/components/EditFiltersModal.tsx:
```diff
@@ -57,19 +57,8 @@ const EditFiltersModal = ({ isMobile = false }) => {
-
     const filters = useAppSelector(selectNeighborhoodFilters);
-
-    useEffect(() => {
```
Great to see the initialization in the useState! Just noting here, that in some cases I worked on before, this useEffect might still be needed. We will test the workflows later when they are all in and then we could see if this is necessary or not. Even if we need to add it back, it should be quick and easy.
